### PR TITLE
feat(arm): AzureDefenderOnSqlServersVMS - checks resource  is a SQL with tier "Standard".

### PR DESCRIPTION
**Pull request by unu87**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
The function scan_resource_conf checks if a specific resource is a SQL Server with tier "Standard". If so, it returns a failure result; otherwise, it returns a success result.

### Fix
The function checks if a resource is a SQL Server Virtual Machine with tier "Standard" and returns a pass or fail result accordingly.

## Checklist:

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ x] I have added tests that prove my feature, policy, or fix is effective and works
- [x ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.arm.base_resource_check import BaseResourceCheck
+from typing import List, Dict, Any
+
+
+class AzureDefenderOnSqlServersVMS(BaseResourceCheck):
+    def __init__(self):
+        name = "Ensure that Azure Defender is set to On for SQL servers on machines"
+        id = "CKV_AZURE_79"
+        supported_resources = ("Microsoft.Cache/Redis",)
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:10]: Probably should be `Microsoft.Security/pricings` type and not `Microsoft.Cache/Redis`

File: checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get("properties", {})
+        resourceType = properties.get("resourceType")
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:21]: I don't think there is `resourceType` in this resource, we can remove it.

Comment (unu87) [reply checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:21]: so how do we det to the "resourceType": "SqlServerVirtualMachines" to test it?

Comment (ChanochShayner) [reply checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:21]: we dont need it for ARM

File: checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get("properties", {})
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:20]:
```suggestion
        properties = conf.get("properties", {})
        name = conf.get("name", "")
```

File: checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        tier = properties.get("tier")
+        if tier == "Standard":
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:22]:
```suggestion
        if tier == "Standard" and name == "SqlServerVirtualMachines":
```

File: tests/arm/checks/resource/example_AzureDefenderOnSqlServersVMS/fail.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+        "type": "Microsoft.Security/pricings",
+        "apiVersion": "2020-01-01",
+        "name": "fail",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_AzureDefenderOnSqlServersVMS/fail.json:8]:
```suggestion
        "name": "SqlServerVirtualMachines",
```

File: tests/arm/checks/resource/example_AzureDefenderOnSqlServersVMS/pass.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+        "type": "Microsoft.Security/pricings",
+        "apiVersion": "2020-01-01",
+        "name": "pass",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_AzureDefenderOnSqlServersVMS/pass.json:8]:
```suggestion
        "name": "SqlServerVirtualMachines",
```

File: tests/arm/checks/resource/test_AzureDefenderOnSqlServersVMS.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        test_files_dir = Path(__file__).parent / "example_AzureDefenderOnSqlServersVMS"
+
+        # when
+        report = Runner().run(root_folder=str(test_files_dir), runner_filter=RunnerFilter(checks=[check.id]))
+
+        # then
+        summary = report.get_summary()
+
+        passing_resources = {
+            "Microsoft.Security/pricings.pass",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/test_AzureDefenderOnSqlServersVMS.py:21]:
```suggestion
            "Microsoft.Security/pricings.SqlServerVirtualMachines",
```

File: tests/arm/checks/resource/test_AzureDefenderOnSqlServersVMS.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        failing_resources = {
+            "Microsoft.Security/pricings.fail",
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/test_AzureDefenderOnSqlServersVMS.py:25]:
```suggestion
            "Microsoft.Security/pricings.SqlServerVirtualMachines",
```

**Review by ChanochShayner — APPROVED** [review]

File: checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            name=name,
+            id=id,
+            categories=categories,
+            supported_resources=supported_resources,
+        )
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get("properties", {})
+        name = conf.get("name", "")
+        tier = properties.get("tier")
```
Comment (SteveVaknin) [inline checkov/arm/checks/resource/AzureDefenderOnSqlServersVMS.py:22]: Any reason one has default value and one doesn't?   
If not please follow one standard

**Review by SteveVaknin — APPROVED** [review]
Looks good, please notice one comment

**Pull request merged**